### ISSUE TITLE
Hook into external metric libraries

### ISSUE THREAD

**Issue by phinate** [alan-turing-institute/cloudcasting#56]
(Previously discussed in #11)

Libraries like [pix](https://github.com/google-deepmind/dm_pix/blob/master/dm_pix/_src/metrics.py#L65%23L83) already have implementations for all metrics we have + some more, and are backed by JAX for speed. If we could conform to their convention of providing images with rank `NHWC = [number_of_images, height, width, channels]` -- either by going to/from that in our functions, or changing them everywhere -- we could benefit from the speed and additional metrics provided.

N.B. we would lose per-channel info if we piled everything into the last dimension (the metrics average there). We could add a superficial dimension for channel with `np.newaxis` in the last position (I believe this is how they handle greyscale images), and then `vmap` the metrics over both batch and channel, for instance, and that would likely preserve the per-channel metrics.

**Comment by dfulu** [comment]
I did a speed test on our metrics with some dummy data - results below. 

```
from cloudcasting.metrics import mae_batch, mse_batch, ssim_batch
import numpy as np

# shape [batch size, channels, time, y, x]
y_a = np.random.rand(2, 11, 12, 372, 614)
y_b = np.random.rand(2, 11, 12, 372, 614)

%timeit -n 2 -r 2 _ = mae_batch(y_a, y_b)
>> 821 ms ± 82.4 ms per loop (mean ± std. dev. of 2 runs, 2 loops each)

%timeit -n 2 -r 2 _ = mse_batch(y_a, y_b)
>> 736 ms ± 7.78 ms per loop (mean ± std. dev. of 2 runs, 2 loops each)

%time _ = ssim_batch(y_a, y_b)
>> CPU times: user 4.5 s, sys: 305 ms, total: 4.81 s
>> Wall time: 4.81 s
```

So ignoring the speed to serve batches to the main process for the moment, our validation set has ~15k samples, so 7.k batches of batch size 2 as used above. In the loop we'd expect these total times spent calculating results for each metric:

- MAE: ~1.7 hours
- MSE: ~1.5 hours
- SSIM: ~10 hours

So including SSIM as we have it now will add 10 hours to whatever the runtime is to validate a model. I think Evie's estimated that when using 0 workers in the dataloader (so slow serving of batches), validating her model would have taken 40 hours - that was with only MAE and MSE

**Comment by phinate** [comment]
Closed by #58.
